src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -91,12 +99,52 @@ static ImmutableHashSet<IOption> ComputeSerializableOptionsFromProviders(Immutab
             }
         }
 
+        private ImmutableArray<IOptionPersister> GetOptionPersisters()
```

Comment:
should this become async/cancellable in teh future?

Reply:
Yes, but it is a _much_ more involved change. I wouldn't expect it to happen unless it became required as a result of some independent observation of a problem.